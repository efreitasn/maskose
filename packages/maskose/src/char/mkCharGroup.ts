import { MaskCharPredicateFn, MaskChar } from '.';

export const MASK_CHAR_GROUP_TYPE = 'MASK_CHAR_GROUP_TYPE';

export type MaskCharGroup = {
  type: typeof MASK_CHAR_GROUP_TYPE;
  maskChars: MaskChar[];
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A group of characters
 * @param maskChars The characters that will be in the group
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharGroup(
  maskChars: MaskChar[],
  predicateFn?: MaskCharPredicateFn
): MaskCharGroup {
  return {
    type: MASK_CHAR_GROUP_TYPE,
    maskChars,
    predicateFn
  };
}