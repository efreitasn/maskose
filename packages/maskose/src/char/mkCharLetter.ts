import { MaskCharPredicateFn } from '.';

export const MASK_CHAR_LETTER_TYPE = 'MASK_CHAR_LETTER_TYPE';

export type MaskCharLetterObj = {
  type: typeof MASK_CHAR_LETTER_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A character in the range A to Z (case insensitive) expected to be in the value to be masked
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharLetter(predicateFn?: MaskCharPredicateFn): MaskCharLetterObj {
  return {
    type: MASK_CHAR_LETTER_TYPE,
    predicateFn
  };
}