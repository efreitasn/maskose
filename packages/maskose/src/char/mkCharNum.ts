import { MaskCharPredicateFn } from '.';

export const MASK_CHAR_NUM_TYPE = 'MASK_CHAR_NUM_TYPE';

export type MaskCharNumObj = {
  type: typeof MASK_CHAR_NUM_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A character in the range 0 to 9 expected to be in the value to be masked
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharNum(predicateFn?: MaskCharPredicateFn): MaskCharNumObj {
  return {
    type: MASK_CHAR_NUM_TYPE,
    predicateFn
  };
}