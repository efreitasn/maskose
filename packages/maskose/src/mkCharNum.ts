const MASK_CHAR_NUM_TYPE = 'MASK_CHAR_NUM_TYPE';

export type MaskCharPredicateFn = (arg: {
  value: string;
}) => boolean;

export type MkCharNumObj = {
  type: typeof MASK_CHAR_NUM_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

export default function mkCharNum(predicateFn?: MaskCharPredicateFn): MkCharNumObj {
  return {
    type: MASK_CHAR_NUM_TYPE,
    predicateFn
  };
}