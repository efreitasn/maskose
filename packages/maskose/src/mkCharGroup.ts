import { MaskCharPredicateFn, MkCharNumObj } from './mkCharNum';
import { MkCharLetterObj } from './mkCharLetter';
import { MkCharToBePutObj } from './mkCharToBePut';
import { MkCharSpecificObj } from './mkCharSpecific';

const MASK_CHAR_GROUP_TYPE = 'MASK_CHAR_GROUP_TYPE';

export type CharObj =
  | MkCharLetterObj
  | MkCharSpecificObj
  | MkCharNumObj
  | MkCharToBePutObj
;

export type MkCharGroupObj = {
  type: typeof MASK_CHAR_GROUP_TYPE;
  charObjs: CharObj[];
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A group of characters
 * @param charObjs The characters that will be in the group
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharGroup(charObjs: CharObj[], predicateFn?: MaskCharPredicateFn): MkCharGroupObj {
  return {
    type: MASK_CHAR_GROUP_TYPE,
    charObjs,
    predicateFn
  };
}