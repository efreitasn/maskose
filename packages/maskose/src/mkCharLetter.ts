import { MaskCharPredicateFn } from './mkCharNum';

const MASK_CHAR_LETTER_TYPE = 'MASK_CHAR_LETTER_TYPE';

export type MkCharLetterObj = {
  type: typeof MASK_CHAR_LETTER_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

export default function mkCharLetter(predicateFn?: MaskCharPredicateFn): MkCharLetterObj {
  return {
    type: MASK_CHAR_LETTER_TYPE,
    predicateFn
  };
}