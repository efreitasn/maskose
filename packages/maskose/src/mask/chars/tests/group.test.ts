import test from 'ava';
import mkCharGroup from '../group';
import mkCharNum from '../num';
import mkCharLetter from '../letter';
import mkCharBoostRepeat from '../../../boosts/char/repeat';

test('should match the object returned by mkCharGroup()', t => {
  const result = mkCharGroup([
    mkCharBoostRepeat(3)(mkCharNum()),
    mkCharLetter()
  ]);

  t.snapshot(result);
});