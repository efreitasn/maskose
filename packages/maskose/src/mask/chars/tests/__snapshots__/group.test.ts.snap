// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the object returned by mkCharGroup() 1`] = `
Object {
  "chars": Array [
    Object {
      "regExp": /\\[0-9\\]/,
      "repetitions": 3,
      "type": "MASKOSE_CHAR_NUM_TYPE",
      "valueLengthConditions": Array [],
    },
    Object {
      "regExp": /\\[A-Z\\]/i,
      "repetitions": 1,
      "type": "MASKOSE_CHAR_LETTER_TYPE",
      "valueLengthConditions": Array [],
    },
  ],
  "repetitions": 1,
  "type": "MASKOSE_CHAR_GROUP_TYPE",
  "valueLengthConditions": Array [],
}
`;
